File: app/src/main/java/datanapps/androidutility/utils/DNAAppUtils.java
package datanapps.androidutility.utils;

import android.content.Context;
import android.widget.Toast;

import datanapps.androidutility.BuildConfig;



/*
 *
 * Yogendra
 * 11/01/2019
 *
 * */


public class DNAAppUtils {


    /*
     * This included because, sonar raise create bug each class should have constructor
     * */

    DNAAppUtils() {
        // nothing to do here
    }

    /*
    * Application Version name
    * */
    public static String getAppVersionName() {
        return BuildConfig.VERSION_NAME;
    }

    /*
     * Application Version Code
     * */
    public static int getAppVersionCode() {
        return BuildConfig.VERSION_CODE;
    }



    /*
    *
    * Application package name and id both are same
    * */
    public static String getApplicationId() {
        return BuildConfig.APPLICATION_ID;
    }


    /*
     *
     * Application package name
     * */
    public static String getApplicationPackageName() {
        return getApplicationId();
    }



    /*
     *
     * Application build type
     * */
    public static String getBuildType() {
        return BuildConfig.BUILD_TYPE;
    }


    /*
     *
     * Application build type
     * */
    public static boolean isBuildType() {
        return BuildConfig.DEBUG;
    }

}

File: app/src/main/java/datanapps/androidutility/utils/kotlin/DNAAppUtils.kt
package datanapps.androidutility.utils.kotlin

import datanapps.androidutility.BuildConfig


/*
 *
 * Yogendra
 * 11/01/2019
 *
 * */


object DNAAppUtils {
/*
     * This included because, sonar raise create bug each class should have constructor
     * */



        /*
    * Application Version name
    * */
        val appVersionName: String
            get() = BuildConfig.VERSION_NAME

        /*
     * Application Version Code
     * */
        val appVersionCode: Int
            get() = BuildConfig.VERSION_CODE


        /*
    *
    * Application package name and id both are same
    * */
        val applicationId: String
            get() = BuildConfig.APPLICATION_ID


        /*
     *
     * Application package name
     * */
        val applicationPackageName: String
            get() = applicationId


        /*
     *
     * Application build type
     * */
        val buildType: String
            get() = BuildConfig.BUILD_TYPE


        /*
     *
     * Application build type
     * */
        val isDebug: Boolean
            get() = BuildConfig.DEBUG


}
